### PR TITLE
Add Tailscale Widget

### PR DESCRIPTION
**Pull request by andrejarrell**
## Proposed change

This change introduces the new Tailscale widget that pulls stats from Tailscale's API.

Screen Shot 2023-05-06 at 10 40 27

The user will be able to add the widget to the `services.yaml` file using the following configuration:
```yaml
widget:
    type: tailscale
    deviceid: deviceid
    key: tailscaleapikey
```

The Tailscale API is documented here:
https://github.com/tailscale/tailscale/blob/main/api.md#device

You can create an API access token on this page:
https://login.tailscale.com/admin/settings/keys

To find your device ID go to the Machines page:
https://login.tailscale.com/admin/machines
Select your machine, scroll down to "Machine Details" and copy the `ID`

The API request would look like this:

```http
GET https://api.tailscale.com/api/v2/device/xxxxx
Authorization: Bearer tskey-api-xxxxx
```

The API response would like this but we will only use `addresses`, `lastSeen`, `expires`, `keyExpiryDisabled`:

```json
{
  "addresses": [
    "100.87.74.78",
    "fd7a:115c:a1e0:ac82:4843:ca90:697d:c36e"
  ],
  "name": "pangolin.tailfe8c.ts.net",
  "hostname": "pangolin",
  "clientVersion": "",
  "updateAvailable": false,
  "os": "linux",
  "created": "2022-12-01T05:23:30Z",
  "lastSeen": "2022-12-01T05:23:30Z",
  "keyExpiryDisabled": true,
  "expires": "2023-05-30T04:44:05Z",
  "authorized": true,
  "isExternal": true,
  "machineKey": "",
  "nodeKey": "nodekey:01234567890abcdef",
  "blocksIncomingConnections": false,
  "enabledRoutes" : [
    "10.0.0.0/16",
    "192.168.1.0/24",
  ],
  "advertisedRoutes" : [
    "10.0.0.0/16",
    "192.168.1.0/24",
  ],
  "tags": [
    "tag:golink"
  ],
  "tailnetLockError": "",
  "tailnetLockKey": "",
}
```

## Type of change

- [x] New service widget
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (please explain)

## Checklist:

- [x] If adding a service widget or a change that requires it, I have added a corresponding PR to the [documentation](https://github.com/benphelps/homepage-docs) here: https://github.com/benphelps/homepage-docs/pull/90
- [x] If applicable, I have checked that all tests pass with e.g. `pnpm lint`.
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.

### CODE REVIEW

File: src/widgets/tailscale/component.jsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    if (diffInMinutes > 1) return `${diffInMinutes}m`;
+    const diffInSeconds = Math.ceil(diff / 1000);
+    if (diffInSeconds > 10) return `${diffInSeconds}s`;
+    return "Now";
+  };
+
+  const getLastSeen = () => {
+    const date = new Date(lastSeen);
+    const diff = compareDifferenceInTwoDates(date, now);
+    return `${diff === "Now" ? diff : `${diff} Ago`}`;
```
Comment (shamoon) [inline src/widgets/tailscale/component.jsx:54]: This won't work with translation. You'll need to just return the number or now and in the block call the translation for any other strings like "y" and "Ago"

File: src/widgets/tailscale/component.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import Container from "components/services/widget/container";
+import Block from "components/services/widget/block";
+import useWidgetAPI from "utils/proxy/use-widget-api";
+
+export default function Component({ service }) {
+  const { widget } = service;
+
+  const { data: statsData, error: statsError } = useWidgetAPI(widget, "device", {
+    refreshInterval: 1000 * 60,
```
Comment (shamoon) [inline src/widgets/tailscale/component.jsx:9]: We currently don't allow service widgets to auto-refresh

**Review by shamoon — CHANGES_REQUESTED** [review]
Thanks. Looks good overall and thanks for going through the FR first.

I will test later but noted some minor things here too.

**Comment by andrejarrell** [comment]
@shamoon I have removed the `refreshInterval` and added translations for the time units and for `Ago`. Let me know if this is to your liking.

**Comment by shamoon** [comment]
Cheers thanks for the contribution

**Comment by andrejarrell** [comment]
@shamoon Thanks for adding the `now` and `never`. I was contemplating adding those. (First time working with those translations) I guess I misunderstood the change request for translation. Anyways, thanks for the prompt responses!

**Pull request merged**